components/datetime/src/scaffold/calendar.rs

```diff
@@ -712,18 +712,18 @@ pub trait InSameCalendar {
     ) -> Result<(), MismatchedCalendarError>;
 }
 
-impl<C: Calendar, A: AsCalendar<Calendar = C>> InSameCalendar for Date<A> {
+impl<C: IntoAnyCalendar, A: AsCalendar<Calendar = C>> InSameCalendar for Date<A> {
```

Comment:
Praise: When I saw this PR, I was wondering how you were going to implement this function. It seems clean to just move the trait bound to a different trait that has the needed function.